lib/system/sysstr.nim

```diff
@@ -325,41 +325,46 @@ proc nimIntToStr(x: int): string {.compilerRtl.} =
   result = newStringOfCap(sizeof(x)*4)
   result.add x
 
-proc add*(result: var string; x: float) =
+
+proc writeFloatToBuffer(buf: var array[64, char]; value: BiggestFloat): int =
+  ## returns the amount of bytes written to buffer not counting the '\0' terminating character.
+  var n: int = c_sprintf(buf[0].addr, "%.16g", value)
+  var hasDot = false
+  for i in 0..n-1:
+    if buf[i] == ',':
+      buf[i] = '.'
+      hasDot = true
+    elif buf[i] in {'a'..'z', 'A'..'Z', '.'}:
+      hasDot = true
+  if not hasDot:
+    return n + c_sprintf(buf[n].addr, ".0")
```

Comment:
This and the following ``c_sprintf`` are to be replaced by proper append operations.

Reply:
how should I do that? Which append proc are you referring to? It's not an appendable array. It is a static array. I could write the following: 

```nim    
  buf[n .. n+2] = ".0\0"
  return n + 2
```
But that operator isn't even supported, nor do I think it's better readable. With the sprintf and the index n in the argument I think I make it pretty clear that I want to append the string including the 0-terminator.